compiler/parser/Parser.y

```diff
@@ -1002,17 +1002,39 @@ importdecls_semi
         | {- empty -}           { [] }
 
 importdecl :: { LImportDecl GhcPs }
-        : 'import' maybe_src maybe_safe optqualified maybe_pkg modid maybeas maybeimpspec
-                {% ams (cL (comb4 $1 $6 (snd $7) $8) $
-                  ImportDecl { ideclExt = noExt
-                             , ideclSourceSrc = snd $ fst $2
-                             , ideclName = $6, ideclPkgQual = snd $5
-                             , ideclSource = snd $2, ideclSafe = snd $3
-                             , ideclQualified = snd $4, ideclImplicit = False
-                             , ideclAs = unLoc (snd $7)
-                             , ideclHiding = unLoc $8 })
-                   ((mj AnnImport $1 : (fst $ fst $2) ++ fst $3 ++ fst $4
-                                    ++ fst $5 ++ fst $7)) }
+        : 'import' maybe_src maybe_safe optqualified maybe_pkg modid optqualified maybeas maybeimpspec
+                {% do
+                      -- Note : [Poor man's 'ImportQualifiedPost']
+                      -- -----------------------------------------
+                      -- Knowing we are downhill from the official
+                      -- implementation of 'ImportQualifiedPost'
+                      -- (https://gitlab.haskell.org/ghc/ghc/merge_requests/853)
+                      -- in the interests of saving time and effort, I'm
+                      -- playing fast and loose here. Specifically, I'm
+                      -- skipping adding the extension and its
+                      -- associated warning
+                      -- 'prepositive-qualified-module' and I'm not
```

Comment:
I think we already turn on the warning in DAML, which means its useful/necessary... How much harder is it to add back the warning? I don't care so much about the extension, as we turn it on by default.

Reply:
We do turn on the warning, I made it so. It's not a big deal to add the warning if you think it's important (I don't). Please let me know.

Reply:
Up to you. It was cute to get the warning, but not a huge deal, and we get it with 8.10.

Reply:
I think the warning is a cutie too obviously :) But yes, exactly - it comes back for free in 8.10 and will never be an error so causes no harm if it has a little holiday.

Reply:
Makes sense. Good plan.